predicates/interval/interval.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2015 Zalando SE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+Package interval implements custom predicates to match routes
+only during some period of time. Package includes three predicates:
+Between, Before and After. All predicates can be created using the date
+represented as a string in RFC3339 format (see https://golang.org/pkg/time/#pkg-constants),
+int64 or float64 number. float64 number will be converted into int64
+number.
+
+Between predicate matches only if current date is inside the specified
+range of dates. Range is a closed range, so boundaries are included in
+the range. Between predicate requires two dates to be constructed.
+Upper boundary must be after lower boundary.
+
+Before predicate matches only if current date is before the specified
+date. Only one date is required to construct the predicate. Boundary
+is not included in the range.
+
+After predicate matches only if current date is after the specified
+date. Only one date is required to construct the predicate. Boundary
+is not included in the range.
+
+Examples:
+
+	example1: Path("/zalando") && Between("2016-01-01T12:00:00+02:00", "2016-02-01T12:00:00+02:00") -> "https://www.zalando.de";
+	example2: Path("/zalando") && Between(1451642400, 1454320800) -> "https://www.zalando.de";
+
+	example3: Path("/zalando") && Before("2016-02-01T12:00:00+02:00") -> "https://www.zalando.de";
+	example4: Path("/zalando") && Before(1454320800) -> "https://www.zalando.de";
+
+	example3: Path("/zalando") && After("2016-01-01T12:00:00+02:00") -> "https://www.zalando.de";
+	example4: Path("/zalando") && After(1451642400) -> "https://www.zalando.de";
+
+*/
+package interval
+
+import (
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/zalando/skipper/routing"
+)
+
+type intervalType int
+
+const (
+	between intervalType = iota
+	before
+	after
+)
+
+var InvalidArgsError = errors.New("invalid arguments")
```

Comment:
I was just suggesting to move it to predicate package, not filter or higher.
Besides, it's more idiomatic to name these vars `ErrSomething`